Unix/base/user.c

```diff
@@ -592,6 +594,25 @@ static int GetGroupName(
     return 0;
 }
 
+int GetGroupId(
+    const char *groupName,
+    gid_t *gid)
+{
+    struct group grbuf;
+    char buf[1024];
+    struct group* gr;
+
+    if (getgrnam_r(groupName, &grbuf, buf, sizeof(buf), &gr) != 0)
+        return -1;
```

Comment:
Error message written to stderr in calling function, since logging is not yet set up at this point.

This function is called during parsing of configuration file, so still single-threaded.